src/components/generics/vacation-test-forms/employee-vacation-requests.tsx

```diff
@@ -92,6 +105,68 @@ const RenderEmployeeVacationRequests = ({ persons }: Props) => {
     initializeRequests();
   }, [person]);
 
+  /**
+   * Initializes all vacation request statuses
+   */
+  const initializeRequestStatuses = async () => {
+    try {
+      const vacationRequestStatuses: VacationRequestStatus[] = [];
+      const statusesApi = Api.getVacationRequestStatusApi(accessToken?.access_token);
+
+      await Promise.all(requests.map(async request => {
+        const createdStatuses = await statusesApi.listVacationRequestStatuses({ id: request.id! });
+        createdStatuses.forEach(createdStatus => {
+          vacationRequestStatuses.push(createdStatus);
+        });
+      }));
+
+      setStatuses(vacationRequestStatuses);
+      setLoading(false);
+    } catch (error) {
+      context.setError(strings.errorHandling.fetchVacationDataFailed, error);
+    }
+  };
+
+  useEffect(() => {
+    if (requests.length <= 0) {
+      return;
+    }
+    initializeRequestStatuses();
+  }, [requests]);
+
+  /**
+   * Initializes all the latest vacation request statuses, so there would be only one status for each request showed on the UI
+   */
+  const initializeLatestStatuses = async () => {
+    const latestStatuses: VacationRequestStatus[] = [];
+
+    requests.forEach(request => {
+      const requestStatuses: VacationRequestStatus[] = [];
+  
+      // Get statuses for this particular request
+      statuses.forEach(status1 => {
+        if (request.id === status1.vacationRequestId) {
+          requestStatuses.push(status1);
+        }
+      });
+  
+      // Pick the latest statuses
+      if (requestStatuses.length > 0) {
```

Comment:
Could reduce this to (requestStatuses.length)

Reply:
Reduced it to (requestStatuses.length), works as intended.


```suggestion
      // Pick the latest statuses
      if (requestStatuses.length) {
        ...
      }
```